### PR TITLE
feat: Table Lineage API

### PR DESCRIPTION
**Pull request by allisonsuarez**
Signed-off-by: Allison Suarez Miranda <asuarezmiranda@lyft.com>

### Summary of Changes

- Created Table Lineage API
- Added Column to ResourceType enum
- Bumped version of amundsen-common
- Bumped version of marshmallow and marshmallow-annotations to align with amundsen-common
- Updated tests and uses of marshmallow to new version

### Tests

- Wrote tests for Table Lineage API
- Updated users test to work with marshmallow update
- Manually tested API calls

### Documentation

_What documentation did you add or modify and why? Add any relevant links then remove this line_

### CheckList

Make sure you have checked **all** steps below to ensure a timely review.

- [ ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ ] PR includes a summary of changes.
- [ ] PR adds unit tests, updates existing unit tests, **OR** documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
- [ ] PR passes `make test`

### CODE REVIEW

File: metadata_service/api/swagger_doc/lineage/lineage_get.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  - name: depth
+    in: query
+    type: integer
+    schema:
+      type: integer
+    required: false
+    example: 0
+responses:
+  200:
+    description: 'Lineage for requested driection and depth'
```
Comment (youngyjd) [inline metadata_service/api/swagger_doc/lineage/lineage_get.yml:29]: typo here

File: metadata_service/api/__init__.py [hunk 2]
```diff
@@ -33,11 +33,11 @@ def get_with_kwargs(self, *, id: Optional[str] = None, **kwargs: Optional[Any])
                 actual_id: Union[str, int] = int(id) if id.isdigit() else id
                 object = get_object(id=actual_id, **kwargs)
                 if object is not None:
-                    return self.schema().dump(object).data, HTTPStatus.OK
+                    return self.schema().dump(object), HTTPStatus.OK
```
Comment (feng-tao) [inline metadata_service/api/__init__.py:36]: What is the line about?

File: metadata_service/api/__init__.py [hunk 3]
```diff
@@ -33,11 +33,11 @@ def get_with_kwargs(self, *, id: Optional[str] = None, **kwargs: Optional[Any])
...
                 return None, HTTPStatus.NOT_FOUND
             except ValueError as e:
                 return {'message': f'exception:{e}'}, HTTPStatus.BAD_REQUEST
         else:
             get_objects = getattr(self.client, f'get_{self.str_type}s')
             objects: List[Any] = get_objects()
-            return self.schema(many=True).dump(objects).data, HTTPStatus.OK
+            return self.schema().dump(objects, many=True), HTTPStatus.OK
```
Comment (feng-tao) [inline metadata_service/api/__init__.py:43]: same

File: metadata_service/api/popular_tables.py [hunk 4]
```diff
@@ -26,5 +26,5 @@ def get(self, user_id: Optional[str] = None) -> Iterable[Union[Mapping, int, Non
         limit = request.args.get('limit', 10, type=int)
         popular_tables: List[PopularTable] = self.client.get_popular_tables(num_entries=limit,
                                                                             user_id=user_id)
-        popular_tables_json: str = PopularTableSchema(many=True).dump(popular_tables).data
+        popular_tables_json: str = PopularTableSchema().dump(popular_tables, many=True)
```
Comment (feng-tao) [inline metadata_service/api/popular_tables.py:29]: same

File: metadata_service/api/table.py [hunk 5]
```diff
@@ -31,13 +32,37 @@ def __init__(self) -> None:
     def get(self, table_uri: str) -> Iterable[Union[Mapping, int, None]]:
         try:
             table = self.client.get_table(table_uri=table_uri)
-            schema = TableSchema(strict=True)
-            return schema.dump(table).data, HTTPStatus.OK
+            schema = TableSchema()
```
Comment (feng-tao) [inline metadata_service/api/table.py:35]: I forgot about the context, but why do we remove the strict=true?

File: metadata_service/api/__init__.py (hunk 2 above)
Comment (feng-tao) [reply metadata_service/api/__init__.py:36]: is it because we upgrade the mashmallow lib which the API changes?

Comment (allisonsuarez) [reply metadata_service/api/__init__.py:36]: Yes, as said on the description we updated marshmallow to be in line with the amundsen-common version

File: metadata_service/api/table.py (hunk 5 above)
Comment (allisonsuarez) [reply metadata_service/api/table.py:35]: Similar reason as removing data this wasn't compatible with the update

Comment (allisonsuarez) [reply metadata_service/api/table.py:35]: https://marshmallow.readthedocs.io/en/stable/upgrading.html#schemas-are-always-strict

**Review by feng-tao — APPROVED** [review]

File: setup.py [hunk 6]
```diff
@@ -22,8 +22,14 @@
     packages=find_packages(exclude=['tests*']),
     include_package_data=True,
     zip_safe=False,
-    dependency_links=[],
-    install_requires=requirements,
+    dependency_links=[
+        ('git+https://www.github.com/hilearn/marshmallow-'
+         'annotations.git@a7a2dc96932430369bdef36555082df990ed9bef#egg=marshmallow-annotations')
+    ],
+    install_requires=[
```
Comment (verdan) [inline setup.py:29]: @allisonsuarez is there a reason we removed the requirements file from the `install_requires`? as it breaks the projects if we simply install metadata using setup.py, and hence via pip. Or it's just a typo? 

cc: @feng-tao

**Pull request merged**